cli/main.go

```diff
@@ -296,6 +296,14 @@ func beforeSubcommands(c *cli.Context) error {
 		ignoreConfigLogs = true
 	} else {
 		if path := c.GlobalString("log"); path != "" {
+			// avoid log.json file too large consume the memory footprint in the tmpfs,
+			// truncate the log.json file every time before kata subcommand is executed
+			if path != "/dev/null" && katautils.FileExists(path) {
+				if err := os.Truncate(path, 0); err != nil {
```

Comment:
I'd prefer to have a limit (2 MB maybe) instead of truncate it when a subcommand with --log is called, wdyt?

Reply:
I think have a file limit is not appropriate. If the log file size exceeds the limit, just return the log file exceed the limit error is not a good way.

I find the log file specified with --log is used only in the `getLastRuntimeError` function in the containerd-shim, this function just opens the log file and returns the last error level log message in the log file. when a new kata subcommand is executed, the log message written before is not userless in the log file, so why not truncate it.

Reply:
Does containerd ever set `--log` argument? I cannot find the code that sets it.

Reply:
When `--log` is specified, we should honor `--debug` to set kata's log level as well. Then we mirror the behavior of runc and would not write many log entries there when debug is not enabled.

Reply:
I agree with you. kata-runtime will not produce too much log at the most time. And I think who use this log file who should process this log file(maybe should truncate it), so I recommend to do this modification in the containerd instead of kata-runtime.

I will close this issue and PR.